### ISSUE TITLE
CreateTearsheetStep: Updating props.secondaryLabel causes CreateTearsheet to re-render completely

### ISSUE THREAD

**Issue by janhassel** [carbon-design-system/ibm-products#5875]
### Package

Carbon for IBM Products

### Description

When `props.secondaryLabel` of a `CreateTearsheetStep` is updated, the entire `CreateTearsheet` component re-renders and falls back to the first step.

### Component(s) impacted

CreateTearsheet

### Browser

Safari

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

2.47.0

### Suggested Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Product/offering

watsonx

### CodeSandbox or Stackblitz example

https://stackblitz.com/edit/github-ggvwco-pz8dxm?file=src%2FExample%2FExample.jsx,package.json

### Steps to reproduce the issue (if applicable)

In the stackblitz: click "next" to access the second step; then click "increment"

https://github.com/user-attachments/assets/5944ee44-482f-4d2a-aa58-68ae65878af4

The stackblitz seems to reload the page shortly after somehow causing the tearsheet to disappear and re-appear. This **does not** happen in my actual codebase. Only the re-render of the CreateTearsheet component and therefore loss of navigation.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues

**Comment by davidmenendez** [comment]
hey @janhassel wanted to follow back up with you here. i see behavior in your stackblitz, but i think this is just an issue with implementation. in your example you're using just a regular html `button` and the `onClick` causes the page to submit, so it re-renders.

if you update your handler to include `preventDefault` this behavior doesn't happen and the label updates without causing you to start back at the beginning.

```
onClick={(e) => {
  e.preventDefault();
  setCounter(counter + 1);
}}
```

let me know if this helps or not because i am not able to cause any resetting to happen when updating `secondaryLabel` so i may need your help if this behavior persists.

**Comment by janhassel** [comment]
@davidmenendez Well that explains the page refresh happening on stackblitz vs. no refresh in my local setup.

However, changing the example to a checkbox demonstrates the issue:

https://stackblitz.com/edit/github-ggvwco-a7q5kx?file=src%2FExample%2FExample.jsx,package.json

https://github.com/user-attachments/assets/1ada9fe4-e14d-47e3-9cbd-7bca0305cb56

**Comment by davidmenendez** [comment]
@janhassel thanks for the follow up. luckily for you we just issued a fix for this exact problem in our latest version. it's still a release candidate, but if you take your example and update to [v2.49.0-rc.0](https://github.com/carbon-design-system/ibm-products/releases/tag/v2.49.0-rc.0) you'll see the problem is fixed.

**Comment by janhassel** [comment]
@davidmenendez Yeah, it seems to be fixed with 2.49.0-rc.0. Thanks!
